base/reflection.jl

```diff
@@ -883,7 +888,16 @@ function code_typed(@nospecialize(f), @nospecialize(types=Tuple); optimize=true)
     params = Core.Compiler.Params(world)
     for x in _methods(f, types, -1, world)
         meth = func_for_method_checked(x[3], types)
-        (code, ty) = Core.Compiler.typeinf_code(meth, x[1], x[2], optimize, params)
+        argtypes = nothing
+        if constvals !== nothing
+            argtypes = Vector(undef, length(types.parameters)+1)
+            for i = 1:length(types.parameters)+1
+                argtypes[i] = isassigned(constvals, i) ?
+                    Core.Compiler.Const(constvals[i]) : (
+                        i == 1 ? typeof(f) : types.parameters[i-1])
```

Comment:
There appear to be a couple issues with this (invalid assumptions about the structure of `types`, `typeof` instead of `Core.Typeof`). Would it be possible to format this better too? The deeply nested conditionals and indentation are hard to follow.